vault/logical_system_integ_test.go

```diff
@@ -604,3 +606,137 @@ func TestBackend_PluginMainCredentials(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestSystemBackend_InternalUIResultantACL(t *testing.T) {
+	cluster := vault.NewTestCluster(t, nil, &vault.TestClusterOptions{
+		HandlerFunc: vaulthttp.Handler,
+	})
+	cluster.Start()
+	defer cluster.Cleanup()
+	client := cluster.Cores[0].Client
+
+	resp, err := client.Auth().Token().Create(&api.TokenCreateRequest{
```

Comment:
[Optional] Can we add more policies outside of `default` (possibly from identity too) and see if that is added to the result?

Reply:
I think we should test by using a token that has more than one ACL policy with some overlapping permissions on the same path. I am not sure if `capabilities = append(capabilities, ...)` would end up appending the same capability more than once in that case (i.e. having `"capabilities": ["read", "read"]` for a path if the token has distinct policies with overlapping permissions on the same path).

Reply:
Actually that might not happen since we instantiate with `capabilities := []string{}` on every walk, but does that mean that the last read on that path wins due to `res["capabilities"] = capabilities`? if the token also had a policy `my-custom-policy` on path `"auth/token/lookup-self"` with `"capabilities": ["update"]`, I think the resultant ACL for the token (containing `default`, and `my-custom-policy`) for that path would be  `"capabilities": ["update"]` (if the walk is done alphabetically) instead of `"capabilities": ["read", "update"]`. Not sure if that's the intended behavior.

Reply:
There will only ever be one entry for any given path.

Reply:
Ahh, I had to look into `NewACL()` to see how the ACL object got constructed. Sounds good then.